nx/test/nx/tensor_test.exs

```diff
@@ -127,4 +127,69 @@ defmodule Nx.TensorTest do
       assert_raise RuntimeError, "not supported", fn -> Nx.random_uniform({2, 2}) end
     end
   end
+
+  describe "access with :all" do
+    test "should access the entire tensor" do
+      t = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+      assert t[[:all]] == t
```

Comment:
@josevalim I'm not sure why this is behaving like this. Shouldn't this only bring the first row?

Reply:
Isn't this conceptually the same as `t[:all]`, therefore taking all rows?

Reply:
Oh right. So all implict dimensions are also ":all", right?

Reply:
Yup!